k8s/src/main/scala/io/buoyant/k8s/resources.scala

```diff
@@ -174,19 +189,22 @@ private[k8s] class NsListResource[O <: KubeObject: TypeReference, W <: Watch[O]:
 }
 
 private[k8s] class NsObjectResource[O <: KubeObject: TypeReference, W <: Watch[O]: TypeReference](
-  val client: Client,
-  listPath: String,
+  parent: Resource,
   objectName: String,
+  maybeListName: Option[String] = None,
```

Comment:
I think I called it that because I wanted to use the identifier `listName` somewhere else, but that may be vestigial – I'll check.